glib/src/object.rs

```diff
@@ -2284,15 +2299,15 @@ impl<T: ObjectType> ObjectExt for T {
     }
 
     #[track_caller]
-    fn set_property<V: ToValue>(&self, property_name: &str, value: V) {
+    fn set_property<V: Into<Value>>(&self, property_name: &str, value: V) {
```

Comment:
This doesn't allow passing e.g. a `&some_object` anymore but will require passing `some_object.clone()`, correct?

Reply:
That will still work, I didn't change this `impl From<&T> for Value`: [value.rs#L615-L620](https://github.com/gtk-rs/gtk-rs-core/blob/11222fe785210ef35667a81c14f86085bdf2d129/glib/src/value.rs#L615-L620)

Reply:
Ah so this doesn't actually change the API in a breaking way, and we don't have to change all the code everywhere? Just that we could if we wanted to have fewer copies

Reply:
Not sure this is the right direction for `set_property()` though. The value is effectively borrowed to the object, so the object implementation will have to do a copy anyway when it gets the value.

In C, `g_object_set()` will only borrow the value to the `GValue`, as an optimization.

Reply:
Yes, this does not require any major code changes :) Using `ToValue` will always create a copy so this change is a strict optimization to remove copies in cases where you pass an owned value. To fully avoid all copies here, we would have to add a lifetime parameter to `ToValue` and `Value` so it can hold borrowed values.

Reply:
Yeah I was thinking of some kind of `BorrowedValue<'a>`. All doable but so much to do! :)